### ISSUE TITLE
[Feature Request] Accessibility issue on expansion panel and TreeView

### ISSUE THREAD

**Issue by xiaoshuw** [vuetifyjs/vuetify#12756]
### Problem to solve
On my recent vuetify project, the accessibility audit review some issues on the expansion panel and TreeView, which is internally implemented with expansion panel. According to https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html, the expansion panel has few issues: (1) the aria-expanded attribute seems not landing on the button element but an outer wrapper div. (2) There is no heading element. (3) there is no aria-control attributes and matching id on the target element.

The issue with TreeView include the above issue but due to lack of mechanism, we are unable to customize any behavior.

### Proposed solution
For the Expansion Panel, the desired improvements are:

1.  'aria-expanded' attribute landing on the button element. 
2. Provide a role='heading' on the eventually generated expansion-panel-header element. Provide a property that allows user to pass in the attribute aria-level.
3. Automatically generate the correct aria-controls and its target

For the tree-view, it should also include the above fix, but perhaps with a root aria-level property. In addition, the aria-label on the expansion/collapse button would ideally read the label.

**Comment by ax-jmckenzie** [comment]
The checkboxes that are rendered in a 'selectable' variant of TreeView should also get some accessibility love. 

e.g.

role="checkbox" aria-checked="false" aria-labelledby="chk1-label"

**Comment by JimSadler** [comment]
@xiaoshuw I am in agreement with this change, however, adding role="heading" to the v-expansion-panel-header is not preferable, as this element is rendered as a button, and the Screen Reader should announce that it is a button, not a header. So if the role="heading" is added to a button, it will not get a button announcement.

**Comment by agreedSkiing** [comment]
Solution for treeview could be to add the role `tree` (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tree_role) for the root element and childelements that containt information or have the aria-aria-expanded attribute the role `treeitem` (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/treeitem_role).
Also if an childelement is not directly under the root then it should also be included in a html element with the role `group` see https://www.w3.org/TR/wai-aria-practices-1.1/examples/treeview/treeview-2/treeview-2a.html for an exampel of a structured tree

**Comment by johnleider** [comment]
This Issue is being closed due to inactivity.

If you have any questions, please reach out to us in our [Discord community](https://community.vuetifyjs.com).
